<a href="https://colab.research.google.com/github/xRUFIAxGit1/lab1/blob/main/firstLec.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

first task

Дан диаметр окружности d. Найти ее длину L = π·d. В качестве
значения π использовать 3.14.

0
1
2
3
4
5
6
7
8
9


In [1]:
import math
import random

d = random.randrange(1,10)
print("Диаметр окружности d = ", d)
print("Пи = ", math.pi)
L = math.pi * d
print("Длина окружности L = ", L)


Диаметр окружности d =  1
Пи =  3.141592653589793
Длина окружности L =  3.141592653589793


. Дана площадь S круга. Найти его диаметр D и длину L окружности,
ограничивающей этот круг, учитывая, что L = 2·π·R, S = π·R
2
. В качестве

значения π использовать 3.14.

In [2]:
import random
import math

print("Пи:", math.pi)
S = random.randrange(1,100)
R = math.sqrt(S / math.pi)
L = 2 * math.pi * R
print("Радиус окружности: ", R)
print("Площадь круга: ", S)
print("Длина окружности: ", L)


Пи: 3.141592653589793
Радиус окружности:  4.583497844237541
Площадь круга:  66
Длина окружности:  28.798966310402623


Найти значение функции y = 4(x−3)6 − 7(x−3)3 + 2 при данном
значении x.

In [3]:
import random

x = random.randrange(-10, 11)
print("x = ",x)
t = (x-3)**3
y = 4*t**2 - 7*t + 2
print("4*({0}-3)^6 - 7*({0}-3)^3 + 2 = {1}".format(x,y))


x =  3
4*(3-3)^6 - 7*(3-3)^3 + 2 = 2


Скорость первого автомобиля V1 км/ч, второго — V2 км/ч, расстоя-
ние между ними S км. Определить расстояние между ними через T часов,

если автомобили первоначально движутся навстречу друг другу. Данное
расстояние равно модулю разности начального расстояния и общего пути,
проделанного автомобилями; общий путь = время · суммарная скорость.

In [4]:
import random

V1,V2 = [random.randrange(50,70) for i in range(2)]
S = random.randrange(400,600)
T = random.randrange(1,6)
total_speed = V1 + V2
total_path = total_speed * T
D = abs(S - total_path)

print("1st vehicle speed:",V1)
print("2nd vehicle speed:",V2)
print("Time:",T)
print("Total speed:",total_speed)
print("Initial Distance between vehicles:",S)
print("Final Distance between vehicles:",D)


1st vehicle speed: 65
2nd vehicle speed: 58
Time: 5
Total speed: 123
Initial Distance between vehicles: 469
Final Distance between vehicles: 146


Даны целые положительные числа A и B (A > B). На отрезке дли-
ны A размещено максимально возможное количество отрезков длины B

(без наложений). Используя операцию деления нацело, найти количество
отрезков B, размещенных на отрезке A.

In [5]:
import random

A = random.randrange(1,100)
B = random.randrange(1,A)
print("A = ", A)
print("B = ", B)
x = int(A/B)
print("Количество полных B в A: ", x)



A =  16
B =  14
Количество полных B в A:  1


Дано трехзначное число. Вывести число, полученное при переста-
новке цифр сотен и десятков исходного числа (например, 123 перейдет в

213).

In [6]:
import random

N = random.randrange(100,1000)
print("Число: ", N)
d2 = int(N/100)
d1 = int((N-d2*100)/10)
d0 = N%10
print("Сотни: ", d2)
print("Десятки: ", d1)
print("Единицы: ", d0)
print("Другое число: ", d1*100 + d2*10 + d0)



Число:  558
Сотни:  5
Десятки:  5
Единицы:  8
Другое число:  558


Дни недели пронумерованы следующим образом: 1 — понедель-
ник, 2 — вторник, . . . , 6 — суббота, 7 — воскресенье. Дано целое число K,

лежащее в диапазоне 1–365. Определить номер дня недели для K-го дня
года, если известно, что в этом году 1 января было вторником.

In [7]:
import random

week_days = {
    1: 'понедельник',
    2: 'вторник',
    3: 'среда',
    4: 'четверг',
    5: 'пятница',
    6: 'суббота',
    7: 'воскресенье'
}

K = random.randrange(1,366)
K = 29

i = 1%7 + 1
print("1-е января: ", 1)
print("Номер дня недели: ", i)
print("День недели:",week_days[i])

i = K%7 + 1
print()
print("Номер дня года: ", K)
print("Номер дня недели: ", i)
print("День недели:",week_days[i])


1-е января:  1
Номер дня недели:  2
День недели: вторник

Номер дня года:  29
Номер дня недели:  2
День недели: вторник


Даны два целых числа: A, B. Проверить истинность высказывания:
«Справедливы неравенства A > 2 и B ≤ 3».

In [8]:
import random
A = random.randrange(-10,10)
B = random.randrange(-10,10)
print("A = ", A)
print("B = ", B)
x = (A>2) and (B<=3)
print("A > 2: ", (A>2))
print("B <=3: ", (B<=3))
print("A > 2 и B <=3: ", x)


A =  -4
B =  -1
A > 2:  False
B <=3:  True
A > 2 и B <=3:  False


Даны три целых числа: A, B, C. Проверить истинность высказы-
вания: «Ровно два из чисел A, B, C являются положительными».

In [9]:
import random
A = random.randrange(-10,10)
B = random.randrange(-10,10)
C = random.randrange(-10,10)
print("A = ", A)
print("B = ", B)
print("C = ", C)
a = (A>0)
b = (B>0)
c = (C>0)
x = (a and b and not c) \
    or (not a and b and c) \
    or (a and not b and c)
print("A положительно: ", a)
print("B положительно: ", b)
print("C положительно: ", c)
print("Ровно два из чисел A, B, C являются положительными: ", x)


A =  -2
B =  9
C =  7
A положительно:  False
B положительно:  True
C положительно:  True
Ровно два из чисел A, B, C являются положительными:  True


Даны числа x, y. Проверить истинность высказывания: «Точка с
координатами (x, y) лежит в четвертой координатной четверти».

In [10]:
import numpy as np

x,y = list(np.random.choice(range(-10, 11), 2))
print("x = {0}, y = {1}".format(x,y))

b = (x > 0 and y < 0)
print("Точка лежит в 4-й координатной четверти: ", b)


x = -8, y = -4
Точка лежит в 4-й координатной четверти:  False


Даны координаты двух различных полей шахматной доски x1,

Условный оператор 15

y1, x2, y2 (целые числа, лежащие в диапазоне 1–8). Проверить истинность

высказывания: «Король за один ход может перейти с одного поля на дру-
гое».

In [12]:
import random

x1,y1,x2,y2 = [random.randrange(1, 9) for i in range(0,4)]
x1,y1,x2,y2 = [2,2,2,1]

c = (abs(x1-x2) == 1 and abs(y1-y2) == 1) \
    or (x1 == x2 and abs(y1-y2) == 1) \
    or (y1 == y2 and abs(x1-x2) == 1)

print("The 1st field:")
print("x1: ", x1)
print("y1: ", y1)
print()
print("The 2nd field:")
print("x2: ", x2)
print("y2: ", y2)
print()

print("King can move in one turn \
from one field to another: ",(c))


The 1st field:
x1:  2
y1:  2

The 2nd field:
x2:  2
y2:  1

King can move in one turn from one field to another:  True


Даны три целых числа. Найти количество положительных чисел в исход-
ном наборе.

In [13]:
import random

A = random.randrange(-3,3)
B = random.randrange(-3,3)
C = random.randrange(-3,3)
print("Три числа:", A, B, C)

x = 0
if A > 0:
    x += 1
if B > 0:
    x += 1
if C > 0:
    x += 1

print("Количество положительных чисел: ", x)


Три числа: 1 -2 -3
Количество положительных чисел:  1


Даны три числа. Найти сумму двух наибольших из них.

In [14]:
import random

A = random.randrange(-30,30)
B = random.randrange(-30,30)
C = random.randrange(-30,30)

print("Число A:", A)
print("Число B:", B)
print("Число C:", C)

if (C <= A and A <= B) or (C <= B and B <= A):
    m = A + B
elif (B <= A and A <= C) or (B <= C and C <= A):
    m = A + C
elif (A <= C and C <= B) or (A <= B and B <= C):
    m = B + C

print()    
print("Сумма:", m)


Число A: 0
Число B: -26
Число C: -24

Сумма: -24


Для данного вещественного x найти значение следующей функции f,
принимающей вещественные значения:

−x, если x ≤ 0,

f (x) = x
2
, если 0 < x < 2,
4, если x ≥ 2.

In [15]:
for i in range(-2,7):
    x = i / 2
    if x <= 0:
        y = -x
    elif x < 2:
        y = x * x
    else:
        y = 4
    print("x = {0} : f(x) = {1}".format(x,y))



x = -1.0 : f(x) = 1.0
x = -0.5 : f(x) = 0.5
x = 0.0 : f(x) = -0.0
x = 0.5 : f(x) = 0.25
x = 1.0 : f(x) = 1.0
x = 1.5 : f(x) = 2.25
x = 2.0 : f(x) = 4
x = 2.5 : f(x) = 4
x = 3.0 : f(x) = 4


Дано вещественное число — цена 1 кг конфет. Вывести стоимость 1,
2, . . . , 10 кг конфет.

In [16]:
import random

price = round(random.uniform(0, 100),2)
print('Цена 1 кг конфет: ', price)
print()
for i in range(1,11):
    print("Стоимость ", i, 'кг: {0:.2f}'.format(i * price))


Цена 1 кг конфет:  34.09

Стоимость  1 кг: 34.09
Стоимость  2 кг: 68.18
Стоимость  3 кг: 102.27
Стоимость  4 кг: 136.36
Стоимость  5 кг: 170.45
Стоимость  6 кг: 204.54
Стоимость  7 кг: 238.63
Стоимость  8 кг: 272.72
Стоимость  9 кг: 306.81
Стоимость  10 кг: 340.90


Дано вещественное число A и целое число N (> 0). Найти A в степе-
ни N:

A
N = A·A· . . . ·A

In [17]:
import random

N = random.randrange(1,10)
print('N = ', N)
A = random.randrange(-10,10)
print('A = ', A)

P = 1
for i in range(1,N+1):
    P *= A
    print(i," : ",P)
print("Product = ",P)


N =  7
A =  5
1  :  5
2  :  25
3  :  125
4  :  625
5  :  3125
6  :  15625
7  :  78125
Product =  78125


Дано вещественное число X (|X| < 1) и целое число N (> 0). Найти

Цикл с параметром 23

значение выражения
X − X
3
/3 + X
5
/5 − . . . + (−1)N ·X

2·N+1/(2·N+1).

Полученное число является приближенным значением функции arctg в
точке X.

In [18]:
import math
X = 0.8
N = 20
print('X = ', X)
print('N = ', N)

p = X
S = X
k = 1
for i in range(1,N+1):
    p *= -1*X*X
    k += 2
    S += p/k
    print(i," : ",k," : ",p," : ",S)

print("Result:")
print(S)
print("atan(x):")
print(math.atan(X))


X =  0.8
N =  20
1  :  3  :  -0.5120000000000001  :  0.6293333333333333
2  :  5  :  0.32768000000000014  :  0.6948693333333333
3  :  7  :  -0.20971520000000013  :  0.664910019047619
4  :  9  :  0.13421772800000012  :  0.6798230999365079
5  :  11  :  -0.08589934592000009  :  0.6720140684892352
6  :  13  :  0.05497558138880007  :  0.6762429593652968
7  :  15  :  -0.03518437208883205  :  0.6738973345593746
8  :  17  :  0.022517998136852516  :  0.6752219226850719
9  :  19  :  -0.014411518807585613  :  0.6744634216951989
10  :  21  :  0.009223372036854794  :  0.6749026298874301
11  :  23  :  -0.005902958103587069  :  0.6746459795351002
12  :  25  :  0.003777893186295725  :  0.674797095262552
13  :  27  :  -0.0024178516392292645  :  0.6747075452018398
14  :  29  :  0.0015474250491067296  :  0.6747609046862918
15  :  31  :  -0.0009903520314283071  :  0.6747289578465683
16  :  33  :  0.0006338253001141167  :  0.6747481646738445
17  :  35  :  -0.00040564819207303477  :  0.6747365747254995
18  :

Дано целое число N (> 0). Найти сумму

1
1 + 22 + . . . + N
N .

Чтобы избежать целочисленного переполнения, вычислять слагаемые

этой суммы с помощью вещественной переменной и выводить резуль-
тат как вещественное число.

In [19]:
import random

N = random.randrange(1,11)

print("N = ",N)
print()

s = 0
for i in range(1,N+1):
    p = 1.0
    for j in range(1,i+1):
        p *= i
    s += p
    print("i = ",i)
    print("p = ",p)
    print("s = ",s)
    print()

print("Sum = ",s)



N =  5

i =  1
p =  1.0
s =  1.0

i =  2
p =  4.0
s =  5.0

i =  3
p =  27.0
s =  32.0

i =  4
p =  256.0
s =  288.0

i =  5
p =  3125.0
s =  3413.0

Sum =  3413.0


Дано целое число N (> 0). Если оно является степенью числа 3, то
вывести TRUE, если не является — вывести FALSE.

In [20]:
import random

#N = random.randrange(1,1000)
N = 81
print('N = ', N)

while N >= 3:
    N /= 3
print("Является степенью 3: ", (N==1))


N =  81
Является степенью 3:  True


Начальный вклад в банке равен 1000 руб. Через каждый месяц раз-
мер вклада увеличивается на P процентов от имеющейся суммы (P —

вещественное число, 0 < P < 25). По данному P определить, через сколько

месяцев размер вклада превысит 1100 руб., и вывести найденное количе-
ство месяцев K (целое число) и итоговый размер вклада S (вещественное

число).

In [21]:
import random

S_start = 1000
S_end = 1100

P = random.randrange(10,260)/10
#P = 7
print('P = ', P)
coef = 1 + P/100
print("Initial Amount = {0}, Percent = {1}, Coef = {2}".format(S_start,P,coef))
K = 0
S = S_start
while S < S_end:
    S *= coef
    K += 1
    print("K = {0}, S = {1}".format(K,S))

print()
print("Months = {0}, Final Amount = {1}".format(K,S))


P =  6.8
Initial Amount = 1000, Percent = 6.8, Coef = 1.068
K = 1, S = 1068.0
K = 2, S = 1140.624

Months = 2, Final Amount = 1140.624


Дано целое число N (> 1), являющееся числом Фибоначчи: N = FK
(определение чисел Фибоначчи дано в задании While24). Найти целые
числа FK−1 и FK+1 — предыдущее и последующее числа Фибоначчи.

In [22]:
import random

fib = []

def Fib1(N):
    if N < len(fib):
        #print("Fast")
        return fib[N-1]
    if N == 1 or N == 2:
        if N > len(fib):
            fib.append(1)
        return 1
    #print("Slow")
    y = Fib1(N-2) + Fib1(N-1)
    if N > len(fib):
        fib.append(y)
    return y

K = random.randint(1,40)
#N = 4181
print("K = ",K)
N = Fib1(K)
print("Fibonacci Number (N): ", N)

F1 = F2 = 1
print(1,":",F1)
print(2,":",F2)
i = 2
while F1 < N:
    F0, F1, F2 = F1, F2, F1+F2
    i += 1
    print(i,":",F1)
print()
print("{0} + {1} = {2}".format(F0,N,F2))


K =  34
Fibonacci Number (N):  5702887
1 : 1
2 : 1
3 : 1
4 : 2
5 : 3
6 : 5
7 : 8
8 : 13
9 : 21
10 : 34
11 : 55
12 : 89
13 : 144
14 : 233
15 : 377
16 : 610
17 : 987
18 : 1597
19 : 2584
20 : 4181
21 : 6765
22 : 10946
23 : 17711
24 : 28657
25 : 46368
26 : 75025
27 : 121393
28 : 196418
29 : 317811
30 : 514229
31 : 832040
32 : 1346269
33 : 2178309
34 : 3524578
35 : 5702887

3524578 + 5702887 = 9227465
